### PR TITLE
Fix #5039 Terms of Service and Privacy Policy content should still be left-aligned, even if the app is in an RTL language.

### LINKED ISSUE #5039: [BUG]: Terms of Service and Privacy Policy content should still be left-aligned, even if the app is in an RTL language.
**Describe the bug**

In RTL languages, we are currently keeping our Terms of Service and Privacy policy content in English (so that we can have a canonical version). However, the **English** text should be left-aligned.

(Thanks to @KolliAnitha for the repro and images, which I took from https://github.com/oppia/oppia-android/issues/5028.)

**To Reproduce**
1. Go to 'Home' - Options
2. Select App language as Arabic
3. Go back and click on Help - Terms of Service/ Privacy policy
4. See that the content is right-aligned despite being in English.

**Expected behavior**
The English content should be left-aligned (but not the note at the bottom of the page linking to the up-to-date policy).

**Demonstration**
SS

**Environment**
Device/emulator being used: Infinix SMART 5
Android or SDK version (e.g. Android 5 or SDK 21): Android 11
App version (you can get this through system app settings or via the admin controls menu in-app): 0.11-beta-8c81c98d8b

### 1. Update ForcedActivityLanguageMode Proto

Update the `ForcedActivityLanguageMode` enum in `languages.proto` to include a new option:

```proto
USE_ENGLISH = 3;
```

This will allow activities to explicitly request English-only mode.

### 2. Create InjectableAutolocalizedEnglishOnlyActivity

Create a new `InjectableAutolocalizedEnglishOnlyActivity.kt` following the same pattern as [InjectableAutoLocalizedAppCompatActivity](https://github.com/oppia/oppia-android/blob/develop/app/src/main/java/org/oppia/android/app/activity/InjectableAutoLocalizedAppCompatActivity.kt)

This new activity class will serve as the base class for activities that need to be displayed exclusively in English.

### 3. Add New Function to [TranslationController](https://github.com/oppia/oppia-android/blob/develop/domain/src/main/java/org/oppia/android/domain/translation/TranslationController.kt)

Introduce a new function in [TranslationController](https://github.com/oppia/oppia-android/blob/develop/domain/src/main/java/org/oppia/android/domain/translation/TranslationController.kt):

```kotlin
fun getLocaleFor(language: OppiaLanguage): DataProvider<OppiaLocale.DisplayLocale>
```

This function will be used in [AppLanguageWatcherMixin.kt](https://github.com/oppia/oppia-android/blob/469fe9c1a6b949299f8f4d58051388fed47fba83/app/src/main/java/org/oppia/android/app/translation/AppLanguageWatcherMixin.kt#L71-L75) to instantiate the `InjectableAutolocalizedEnglishOnlyActivity`.
### 4. Modify AppLanguageWatcherMixin

Update [AppLanguageWatcherMixin.kt](https://github.com/oppia/oppia-android/blob/469fe9c1a6b949299f8f4d58051388fed47fba83/app/src/main/java/org/oppia/android/app/translation/AppLanguageWatcherMixin.kt#L71-L75) to handle the new `USE_ENGLISH` mode and integrate with the TranslationController changes.

### 5. Modify InjectableSystemLocalizedAppCompatActivity

In [InjectableSystemLocalizedAppCompatActivity.kt](https://github.com/oppia/oppia-android/blob/develop/app/src/main/java/org/oppia/android/app/activity/InjectableSystemLocalizedAppCompatActivity.kt)
- Replace the boolean `shouldOnlyUseSystemLanguage = true` with an enum
- Add enum fields for:
  - `shouldOnlyUseSystemLanguage`
  - `shouldOnlyUseEnglishLanguage` (needed for the PoliciesActivity)

### 6. Modify PoliciesActivity

Update the `PoliciesActivity` to extend the new base class and override the `initializeMixin()` method:

```kotlin
override fun initializeMixin(appLanguageWatcherMixin: AppLanguageWatcherMixin) {
  appLanguageWatcherMixin.initialize(ForcedActivityLanguageMode.USE_ENGLISH)
}
```

### 7. String Content Processing

Extract raw string content (e.g., `val policyDescription`) using regex to:
- Isolate specific tags (such as `li`)
- Apply necessary formatting
- Set the modified string to the TextView

**Reference implementations:**
- [PR #5286 Files](https://github.com/oppia/oppia-android/pull/5286/files)
- [PR #5212 Files](https://github.com/oppia/oppia-android/pull/5212/files)
- [PR #5573](https://github.com/oppia/oppia-android/pull/5573)

### 8. UI Testing

Create UI tests to verify the implementation:

#### Test Requirements:
- **Text Content Verification**: Ensure all text content is displayed in English
- **Text Alignment Verification**: Confirm text is left-aligned using:

```kotlin
assertThat(effectiveGravity and Gravity.HORIZONTAL_GRAVITY_MASK).isEqualTo(Gravity.LEFT)
```
- To properly validate the `USE_ENGLISH` mode, force the system locale to be set to a non-English language (e.g., Swahili) during the test.
- This change will require updating:
    - The system language tests to verify Swahili behavior.
    - The new app language tests to confirm that USE_ENGLISH correctly overrides the system locale.

## Files to Modify/Create

1. **Modify**: `languages.proto` - Add `USE_ENGLISH = 3` enum option
2. **Create**: `InjectableAutolocalizedEnglishOnlyActivity.kt` - New base activity class
3. **Modify**: `TranslationController.kt` - Add `getLocaleFor()` function
4. **Modify**: `AppLanguageWatcherMixin.kt` - Handle new English-only mode
5. **Modify**: `InjectableSystemLocalizedAppCompatActivity.kt` - Replace boolean with enum
6. **Modify**: `PoliciesActivity.kt` - Extend new base class and override initialization
7. **Modify**: `PoliciesFragmentPresenter.kt` - Extract raw string content, apply formatting and set it to textView.
8. **Create**: New UI test cases for validation

### PR DESCRIPTION
**Pull request by ShubhadeepKarmakar**
## Explanation

  - Fix #5039 
  Under the ListItemLeadingMarginSpan class we have the ulSpan class and the olSpan class which takes displayLocal as an input and then retrieves the direction of the layout using the getLayoutDirection() method to get the bullet drawing position.
That's why I created an object of the DisplayLocaleImp class and overridden the getLayoutDirection() method to pretend it's LTR.
  

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))

## Before:
https://github.com/oppia/oppia-android/assets/99060332/e3981549-293e-4cf6-8816-1e51cc711103

## After:
https://github.com/oppia/oppia-android/assets/99060332/d6e037ce-8019-4f98-95d4-556c5289d34b

## Tablet view
tablet-rtl

## Tablet View in landscape
tablet-rtl-landscape

### CODE REVIEW

**Comment by adhiamboperes** [comment]
Hey @ShubhadeepKarmakar , if this is ready for a review please assign it to me. Otherwise, please assign it to yourself.

YOu can run this command to fix your lint failures from terminal `java -jar /home/runner/oppia-android-tools/ktlint -F --android domain/src/**/*.kt utility/src/**/*.kt data/src/**/*.kt app/src/**/*.kt testing/src/**/*.kt scripts/src/**/*.kt  instrumentation/src/**/*.kt`, and don't forget to add screenshots to the PR.

**Comment by adhiamboperes** [comment]
@ShubhadeepKarmakar is this ready to be reviewed?

**Comment by ShubhadeepKarmakar** [comment]
> @ShubhadeepKarmakar is this ready to be reviewed?

Yeah.

File: app/src/main/java/org/oppia/android/app/policies/PoliciesFragmentPresenter.kt [hunk 1]
```diff
@@ -55,9 +71,34 @@ class PoliciesFragmentPresenter @Inject constructor(
...
+      override fun getLayoutDirection(): Int {
+        return ViewCompat.LAYOUT_DIRECTION_LTR
+      }
+    }
+
+    /**
+     * Ensure that the Terms of Service and Privacy Policy are displayed with a left-aligned layout,
+     * even when the app is set to an RTL language.
+     */
+    if (resourceHandler.getLayoutDirection() == ViewCompat.LAYOUT_DIRECTION_RTL) {
```
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/policies/PoliciesFragmentPresenter.kt:88]: I believe we want to align to the start of the textview?

File: app/src/main/java/org/oppia/android/app/policies/PoliciesFragmentPresenter.kt [hunk 2]
```diff
@@ -55,9 +71,34 @@ class PoliciesFragmentPresenter @Inject constructor(
...
+    ) {
+      override fun getLayoutDirection(): Int {
+        return ViewCompat.LAYOUT_DIRECTION_LTR
+      }
+    }
+
+    /**
+     * Ensure that the Terms of Service and Privacy Policy are displayed with a left-aligned layout,
+     * even when the app is set to an RTL language.
+     */
```
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/policies/PoliciesFragmentPresenter.kt]: This should be a regular comment, not a KDoc.

```
//...
//...
```

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
Thanks @ShubhadeepKarmakar!

I was not able to understand the explanation you gave on the PR description, in relation to the problem and the solution. Could you please edit it and clarify?

Overall, I'm unsure of the approach, so I've brought it up with the team. I'll be able to review this further once that's resolved.

Meanwhile, please feel free to work on something else.

**Comment by adhiamboperes** [comment]
Hi @ShubhadeepKarmakar, Could you PTAL at this feedback?
 
Screenshot 2023-10-24 at 13 22 52

**Comment by adhiamboperes** [comment]
@ShubhadeepKarmakar, are you still working on this PR?

File: app/src/main/java/org/oppia/android/app/policies/PoliciesFragmentPresenter.kt (hunk 2 above)
Comment (ShubhadeepKarmakar) [reply app/src/main/java/org/oppia/android/app/policies/PoliciesFragmentPresenter.kt]: Done

File: app/src/main/java/org/oppia/android/app/policies/PoliciesFragmentPresenter.kt (hunk 1 above)
Comment (ShubhadeepKarmakar) [reply app/src/main/java/org/oppia/android/app/policies/PoliciesFragmentPresenter.kt:88]: Yes, but isn't the screen already set to RTL?

**Pull request closed without merging**